### PR TITLE
[e2e] Make it possible to use Rust's test harness for e2e tests

### PR DESCRIPTION
**Pull request by mcy**
Different approach to #92, which avoids having to reinvent a test harness ourselves.

### CODE REVIEW

File: e2e/run.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+# Run Manticore's e2e tests. See the README.md for more details.
+
+# First, build the e2e binary itself, which acts as the server.
+export MANTICORE_E2E_TARGET_BINARY="$(
+  cargo b -p e2e --message-format=json \
+    | tr '\n' ' ' \
+    | perl -pe 's/^.+"executable":"(.+?)".+$/$1/g'
```
Comment (cfrantz) [inline e2e/run.sh:9]: As much as I'd love to complain about perl, the python solution here is also ugly:

```
python -c "import json,sys;print([x for x in [json.loads(line).get('executable') for line in sys.stdin] if x is not None][0])"

## or

python -c "import json,sys;print(list(filter(bool,[json.loads(line).get('executable') for line in sys.stdin]))[0])"
```

There's probably some nice solution with `jq`, but I don't really know that tool, nor is it ubiquitous.

I'm not suggesting a change, just ranting that perl will never go away.

File: e2e/run.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+# Run Manticore's e2e tests. See the README.md for more details.
+
+# First, build the e2e binary itself, which acts as the server.
+export MANTICORE_E2E_TARGET_BINARY="$(
+  cargo b -p e2e --message-format=json \
```
Comment (cfrantz) [inline e2e/run.sh:7]: `cargo build`

File: e2e/run.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    | tr '\n' ' ' \
+    | perl -pe 's/^.+"executable":"(.+?)".+$/$1/g'
+)"
+
+# Now run the tests!
+cargo t -p e2e -- $@
```
Comment (cfrantz) [inline e2e/run.sh:13]: `cargo test`

File: e2e/src/pa_rot.rs [hunk 4]
```diff
@@ -4,110 +4,63 @@
...
-    }
-
-    pub fn millis(s: &str) -> Result<Duration, ParseIntError> {
-        Ok(Duration::from_millis(s.parse()?))
-    }
-}
-
 /// Options for the PA-RoT.
-#[derive(Debug, StructOpt)]
+#[derive(Debug, serde::Deserialize, serde::Serialize)]
```
Comment (cfrantz) [inline e2e/src/pa_rot.rs:32]: Nice: -structopt +serde for communicating parameters to the child?

**Review by cfrantz — APPROVED** [review]

File: e2e/run.sh (hunk 1 above)
Comment (mcy) [reply e2e/run.sh:16]: Be happy it's not sed.

**Pull request merged**